src/games/behavmixed.imp

```diff
@@ -426,23 +375,28 @@ const T &MixedBehaviorProfile<T>::GetBeliefProb(const GameNode &node) const
 {
   CheckVersion();
   ComputeSolutionData();
-  return m_beliefs[node->GetNumber()];
+  return map_beliefs[node];
 }
 
 template <class T>
 Vector<T> MixedBehaviorProfile<T>::GetPayoff(const GameNode &node) const
+// TODO: remove use of Vector
 {
   CheckVersion();
   ComputeSolutionData();
-  return m_nodeValues.Row(node->GetNumber());
+  Vector<T> ret(node->GetGame()->NumPlayers());
+  for (int i = 1; i <= node->GetGame()->NumPlayers(); i++) {
```

Comment:
For this you can use `std::transform` to get rid of the need to index players by integers.

Reply:
I did the following:

```
  /*
  Vector<T> ret(node->GetGame()->NumPlayers());
  for (int i = 1; i <= node->GetGame()->NumPlayers(); i++) {
    ret[i] = map_nodeValues[node][node->GetGame()->GetPlayer(i)];
  }
  return ret;
  */
  Vector<T> ret(node->GetGame()->NumPlayers());
  std::transform (map_nodeValues[node].begin(), map_nodeValues[node].end(), ret.begin(), 
  						[] (std::pair<GamePlayerRep *, T> const &pair) {return pair.second;});
  return ret;
```

What I am not sure about is why the order that transform goes over the players in the map should be the same as the order in `node->GetGame()->NumPlayers()`?

Reply:
Do the iteration over `Game::GetPlayers()` instead of `map_nodeValues`, and that'll ensure the player order is correct by convention.